cas-server-integration-ehcache/src/main/java/org/jasig/cas/ticket/registry/EhCacheTicketRegistry.java

```diff
@@ -50,7 +51,7 @@
  * @author Andrew Tillinghast
  * @since 3.5
  */
-public final class EhCacheTicketRegistry extends AbstractDistributedTicketRegistry implements InitializingBean {
+public final class EhCacheTicketRegistry extends AbstractCrypticTicketRegistry implements InitializingBean {

```

Comment:
Possibly, yes. The drawback of that approach is, it needs explicit configuration and updates to the documentation so that we highlight configuration for each registry and those that are supported to do encryption. This way, everything is automatic. Your configuration stays exactly the same. No other changes. The only thing you're required to do is define a cipher which works for all registries. 

Composition here has other disadvantages: the wrapper that does encryption must be named `ticketRegistry` as the beanId, and the inner registry must be named something else. Which means all other config must be renamed as well. If you wanted to disable encryption, you'd have to reverse the configuration all together. Way too much work for an adopter.